sdk/file_sdk/XMPFileFormat/xmp_file_format.cpp

```diff
@@ -23,12 +23,17 @@ XMPFileFormat::XMPFileFormat(shared_ptr<IStream> file, const string& extension)
 {
 }
 
-const vector<Tag> XMPFileFormat::ReadTags() {
-  return XMPHelper::GetInstance().GetTags(mFile);
+const vector<pair<std::string, std::string>> XMPFileFormat::GetProperties() {
+  if(!mHasProperties){
```

Comment:
Previously, this was not cached - and now it is?
What are the reason for this change?

Reply:
It was always cached, it was in the base class and now it can’t be there.
I don’t mind getting the properties from the file each time, it will be simpler but you said it should be cached...

Reply:
I don't remember why we decide to cache it.
Is there a use case we thought about?

Reply:
it was cashed in the FileFormat class in GetTags

---
In reply to: [140687612](https://github.com/AzureAD/rms-sdk-for-cpp/pull/168#discussion_r140687612) [](ancestors = 140687612)